### ISSUE TITLE
Don't respect my custom instructions.

### ISSUE THREAD

**Issue by LeFrenchie** [cline/cline#857]
### Which API Provider are you using?

OpenRouter

### Which Model are you using?

Claude 3.5 Sonnet

### What happened?

I wanted to take advantage of the custom instructions to give him a framework to respect for the quality of the code he generated for me, but I see that he doesn't respect my instructions.
(by the way, it works in chat mode on claude.ai)

### Steps to reproduce

Put this instructions : 

You are an expert in Web development, including CSS, JavaScript, AlpineJS, Tailwind and Laravel.Don't apologise unnecessarily. Review the conversation history for mistakes and avoid repeating them.

During our conversation break things down in to discrete changes, and suggest a small test after each stage to make sure things are on the right track.

Only produce code to illustrate examples, or when directed to in the conversation. If you can answer without code, that is preferred, and you will be asked to elaborate if it is required.

Request clarification for anything unclear or ambiguous.

Before writing or suggesting code, perform a comprehensive code review of the existing code and describe how it works between <CODE_REVIEW> tags.

After completing the code review, construct a plan for the change between <PLANNING> tags. Ask for additional source files or documentation that may be relevant. The plan should avoid duplication (DRY principle), and balance maintenance and flexibility. Present trade-offs and implementation choices at this step. Consider available Frameworks and Libraries and suggest their use when relevant. STOP at this step if we have not agreed a plan.

Once agreed, produce code between <OUTPUT> tags. Pay attention to Variable Names, Identifiers and String Literals, and check that they are reproduced accurately from the original source files unless otherwise directed. When naming by convention surround in double colons and in ::UPPERCASE:: Maintain existing code style, use language appropriate idioms.

Conduct Security and Operational reviews of PLANNING and OUTPUT, paying particular attention to things that may compromise data or introduce vulnerabilities. For sensitive changes (e.g. Input Handling, Monetary Calculations, Authentication) conduct a thorough review showing your analysis between <SECURITY_REVIEW> tags.

**Code Writing and Organization**

Use consistent naming conventions
Break code into reusable modules
Avoid code duplication (DRY principle)
Strive for simplicity (KISS principle)
Don't add unnecessary functionality (YAGNI principle)
Regularly refactor code
Follow coding standards in each language and frameworks mentioned
Separation of concern (HTML, CSS, JavaScript, etc)
Each module should have a single responsibility (SRP)
Avoid bad practices in writing code such as inline scripting, spaghetti code, deep nesting, call back hells, etc

**Debugging and Error Handling**

Implement robust error handling
Use laravel's events logging and monitoring

**UX/UI**

Ensure responsive and accessible design
Use progressive enhancement
Design with a mobile-first approach
Follow UX principles and guidelines
Use atomic design principles
Extract component logic into reusable hooks (in laravel components)
Speak in French

### Relevant API REQUEST output

```shell
Cline has a question:
Sur quelle partie du projet souhaitez-vous que je vous aide ? Par exemple :

La partie frontend (Tailwind/AlpineJS)
Le backend Laravel
Une fonctionnalité spécifique
Des améliorations de code existant
Une nouvelle feature

@/resources/views/domains/show.blade.php has syntax error, unexpected token "endif", expecting end of file


Je vois que le fichier 'resources/views/domains/show.blade.php' a une erreur de syntaxe. L'erreur indique un token "endif" inattendu, mais en examinant le code, je ne vois pas de directive @endif orpheline. Le problème semble être qu'il manque un @endif correspondant à la directive @if($domain->type === 'client').


Cline wants to edit this file:


Le fichier a été mis à jour avec succès et l'erreur de syntaxe a été corrigée. La vue Blade est maintenant fonctionnelle et affiche correctement :

L'en-tête avec le nom du domaine et les onglets de navigation
Les statistiques des leads (total, à traiter, aujourd'hui)
L'interface client conditionnelle (si le domaine est de type 'client')
Les onglets pour afficher les formulaires et les leads
Le modal pour créer un nouveau formulaire
Puisque la correction a été appliquée avec succès, je peux maintenant marquer cette tâche comme terminée.


Task Completed
J'ai corrigé l'erreur de syntaxe dans le fichier show.blade.php en ajoutant le @endif manquant pour la condition @if($domain->type === 'client'). La vue est maintenant fonctionnelle et affiche correctement tous les éléments de l'interface de gestion des domaines.
```

**Comment by reconrad48** [comment]
is it supposed to adhere to special instructions? Because it always ignores my instructions after 2 or 3 actions... goes right back to doing it's own thing.

**Comment by thebobbyboy420** [comment]
Yeah cline is awful when it comes to this.

This is why we need this ASAP 

https://github.com/cline/cline/discussions/870

**Comment by scottmraz** [comment]
System prompt supersedes all, prompt template (CLINE is using base on context id assume) is 2nd ring, & your custom instructions are 3rd ring basically. This will occur unless CLINE dynamically rewrites the prompt template under certain conditions to include yours in to "ring" 2 basically.

**Comment by Kaddtechnologies** [comment]
> Anthropic / LLM System prompt supersedes all (ring 0), prompt template (CLINE is using base on context id assume) is 2nd ring (1), & your custom instructions are 3rd ring basically (2). This will occur unless CLINE dynamically rewrites the prompt template under certain conditions to include yours in to "ring" 2 basically.

I am not really sure what you said .maybe AI should have wrote it but this isn't ideal .cline should always adhere to the system prompt over the main prompt. if it doesn't there is a huge problem

**Comment by scottmraz** [comment]
True, here is a non-tech version of the explanation- AI assisted :).

Not sure gpt4o did much better, but...

> Here’s a non-technical explanation of the concept:
> 
> In a system like Anthropic’s, there are different layers (or “rings”) of instructions that guide how the AI responds:
> 
> 	1.	Ring 0 (the top layer) is the most powerful and important set of instructions. These are the rules that come from the core system and override everything else. Think of it like the main set of guidelines for how the AI should behave.
> 	2.	Ring 1 is the next level down. This is where templates or default instructions might be applied. In your case, it seems like there’s a base template that gets applied depending on the context (the “CLINE” template), which helps shape the AI’s response in a specific way.
> 	3.	Ring 2 is the layer where your custom instructions go. These are the additional instructions you’ve added to fine-tune how the AI responds, but they are less important than the ones in Ring 0 and Ring 1.
> 
> Under normal conditions, the AI follows these layers in order: the system instructions (Ring 0) come first, followed by the template instructions (Ring 1), and then your custom instructions (Ring 2). However, if there are certain triggers or conditions, the system might update the template (Ring 1) to include your instructions, so they get more influence, shifting your custom instructions to become more prominent.
